plugin.php

```diff
@@ -84,94 +83,98 @@ function create_initial_json_routes() {
 	/*
 	 * Posts
 	 */
-	$controller = new WP_JSON_Posts_Controller;
-	register_json_route( 'wp', '/posts', array(
-		array(
-			'methods'         => WP_JSON_Server::READABLE,
-			'callback'        => array( $controller, 'get_items' ),
-			'args'            => array(
-				'context'          => array(
-					'default'      => 'view',
+	foreach( get_post_types( array( 'public' => true ) ) as $post_type ) {
+		$controller = new WP_JSON_Posts_Controller;
+		$controller->post_type = $post_type;
+		$base = $post_type . 's';
```

Comment:
@danielbachhuber Appending an 's' will not always result in the plural form, or even make sense, for many other languages.